sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -286,6 +286,15 @@ abstract class Expression extends TreeNode[Expression] {
   override def simpleStringWithNodeId(): String = {
     throw new UnsupportedOperationException(s"$nodeName does not implement simpleStringWithNodeId")
   }
+
+  var functionAlias: String = ""
```

Comment:
It's a bad practice to make the `Expression` mutable for trivial things like this. How about using `RuntimeReplaceable`?

Reply:
Argument's data types are matched in Analysis phase of planning and its optimizer's task to replace `RuntimeReplaceable`, correct me if I'm wrong.
So, optimization rules (`ReplaceExpressions`) will never be applied on queries like `select every('true')`.
https://github.com/apache/spark/blob/85cb388ae3f25b0e6a7fc1a2d78fd1c3ec03f341/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L130